### PR TITLE
Create test_urls.py

### PR DESCRIPTION
**Pull request by parismita**
issue #337

### CODE REVIEW

**Comment by parismita** [comment]
what has to be done with the last url??

**Comment by anweshknayak** [comment]
See this (https://github.com/Cloud-CV/EvalAI/blob/master/tests/unit/challenges/test_views.py#L420)
self.challenge_time='PAST' would do

File: tests/unit/challenges/test_urls.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+from django.urls import reverse_lazy
+from rest_framework import status
+from rest_framework.test import APITestCase, APIClient
+from challenges.models import Challenge, ChallengePhase
+from participants.models import Participant, ParticipantTeam
+from hosts.models import ChallengeHost, ChallengeHostTeam
+from base.models import TimeStampedModel
+from django.contrib.auth.models import User
+import unittest
```
Comment (TroJan) [inline tests/unit/challenges/test_urls.py]: can you please reorder your imports according to the following list
* python packages(here unittest)
* django packages(here django.urls)
* thirdy party django packages(here rest_framework)
* project import like apps etc(here challenges, participants, etc)

Also leave a blank line after each set of group and make sure that these imports are written alphabetically.

File: tests/unit/challenges/test_urls.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        url = reverse_lazy('challenges:get_challenge_phase_detail',
+                           kwargs={'challenge_pk': self.challenge.pk, 'pk': self.challenge.pk})
+        self.assertEqual(url, '/api/challenges/challenge/' + str(self.challenge.pk) + '/challenge_phase/' +
+                         str(self.challenge.pk))
+
+        url = reverse_lazy('challenges:get_challenge_by_pk', kwargs={'pk': self.challenge.pk})
+        self.assertEqual(url, '/api/challenges/challenge/' + str(self.challenge.pk) + '/')
+
+        '''url = reverse_lazy('challenges:get_all_challenges', kwargs={'challenge_time': self.challenge_time})
```
Comment (TroJan) [inline tests/unit/challenges/test_urls.py]: can you please remove this comment.

File: tests/unit/challenges/test_urls.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        self.participant_team = ParticipantTeam.objects.create(
+            team_name='Participant Team for Challenge',
+            created_by=self.user)
+
+        self.client.force_authenticate(user=self.user)
+
+
+class TestStringMethods(BaseAPITestClass):
+    def test_challenges_urls(self):
```
Comment (TroJan) [inline tests/unit/challenges/test_urls.py]: can you please add a new line here ?

File: tests/unit/challenges/test_urls.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            anonymous_leaderboard=False)
+
+        self.participant_team = ParticipantTeam.objects.create(
+            team_name='Participant Team for Challenge',
+            created_by=self.user)
+
+        self.client.force_authenticate(user=self.user)
+
+
+class TestStringMethods(BaseAPITestClass):
```
Comment (TroJan) [inline tests/unit/challenges/test_urls.py]: Can we rename it to `TestChallengeUrls` instead of `TestStringMethods`

**Comment by TroJan** [comment]
@parismita : let me know when the pr is ready, so that i can merge it.

**Comment by parismita** [comment]
@TroJan  what does coverall exactly do? Is it failing because earlier it was 83.476 and now its 82.5?
how can i increase it?

**Comment by TroJan** [comment]
@parismita : coveralls(as the name indictes) tells how much of the covered by tests.
In your case, this is because your branch `patch-3` and master have diverged. You can ignore it for now.

But can you please fix the changes which I have suggested ?

**Comment by parismita** [comment]
@TroJan yeah I have fixed what u have said

File: tests/unit/challenges/test_urls.py (hunk 1 above)
Comment (TroJan) [inline tests/unit/challenges/test_urls.py]: the import statements are still not ordered. Can you please order them?

File: tests/unit/challenges/test_urls.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
+from challenges.models import Challenge
```
Comment (TroJan) [inline tests/unit/challenges/test_urls.py]: please move this set of import statements after rest framework import. Order should be like

```
from django.contrib......
from django.urls..........

from rest_framework.test ...........

from challenges.models ........
from hosts.models .........
from participants.models ........
```

**Comment by parismita** [comment]
@TroJan  yeah now the checks are passsing

**Comment by TroJan** [comment]
@parismita : thanks :)

**Pull request merged**